src/main/kotlin/dk/cachet/carp/webservices/study/service/impl/RecruitmentServiceWrapper.kt

```diff
@@ -68,19 +69,22 @@ class RecruitmentServiceWrapper(
             account.carpClaims?.intersect(claims)?.isEmpty() ?: false
         }
 
-    override suspend fun getParticipants(studyId: UUID): List<Account> =
+    override suspend fun getParticipants(
+        studyId: UUID,
+        offset: Int,
+        limit: Int,
+    ): List<Account> =
         withContext(Dispatchers.IO + SecurityCoroutineContext()) {
-            val participants = core.getParticipants(studyId)
+            var participants = core.getParticipants(studyId)
             val accounts = arrayListOf<Account>()
+            if (offset >= 0 && limit > 0) {
+                participants = participants.drop(offset * limit).take(limit).toMutableList()
+            }
+
             for (participant in participants) {
                 val account = accountService.findByAccountIdentity(participant.accountIdentity)
-                if (account != null) {
-                    accounts.add(account)
-                } else {
-                    accounts.add(Account.fromAccountIdentity(participant.accountIdentity))
-                }
+                accounts.add(account ?: Account.fromAccountIdentity(participant.accountIdentity))
```

Comment:
This will work for now, but create an issue for doing the pagination on a database level.